src/engine.ml

```diff
@@ -260,107 +259,55 @@ let client ?pkcs12_password config ts now rng =
         Ok (`Connect (ip, port, dp), Connecting (0, current_ts, 0))
     | [] -> Error (`Msg "couldn't find remote in configuration")
   in
-  match
-    (tls_auth config, tls_crypt config, tls_crypt_v2 config, secret config)
-  with
-  | Error e, Error _, Error _, Error _ -> Error e
-  | Error _, Error _, Error _, Ok (my_key, my_hmac, their_key, their_hmac) ->
-      (* in static key mode, only CBC is allowed *)
-      assert (Config.get Cipher config = `AES_256_CBC);
-      let keys =
+  let* control_crypto =
```

Comment:
Yes! It is nice that this is factored out and we don't have to repeat the other initialization code!